src/Phinx/Console/Command/Create.php

```diff
@@ -52,6 +58,13 @@ protected function configure()
                 PHP_EOL,
                 PHP_EOL
             ));
+
+        // An alternative template.
+        $this->addOption('template', 't', InputOption::VALUE_REQUIRED, 'Use an alternative template');
+
+        // A classname to be used to gain access to the template content as well as the ability to
+        // have a callback once the migration file has been created.
+        $this->addOption('class', 'l', InputOption::VALUE_REQUIRED, 'Use a class implementing "' . self::CREATION_INTERFACE . '" to generate the template');
```

Comment:
Is suppose the issue is do we want to allow a Create subclass to be able to dictate the interface to be used, that interface would have to extend CreationInterface and the subclass would have to provide its own checks to make sure that the class was implementing the appropriate interface. In that case self is still the appropriate choice as as far as the Create class is concerned, it is going to call a method defined in CreationInterface. Hmm. I don't think I'm explaining myself very well. Basically I think self:: is fine, even if a subclass does implement its own value for the constant - it won't be overloaded.